src/engine/runtime.js

```diff
@@ -1824,6 +1828,31 @@ class Runtime extends EventEmitter {
         return varNames;
     }
 
+    /**
+     * Get the label or label function for an opcode
+     * @param {string} extendedOpcode - the opcode you want a label for
+     * @return {object} - object with label and category
+     * @property {string} category - the category for this opcode
+     * @property {Function} [labelFn] - function to generate the label for this opcode
+     * @property {string} [label] - the label for this opcode if `labelFn` is absent
+     */
+    getLabelForOpcode (extendedOpcode) {
+        const [category, opcode] = extendedOpcode.split('_');
```

Comment:
Hmm... `category_opcode` is indeed enforced (see [here](https://github.com/LLK/scratch-vm/pull/1565/files/8602d5798a4a95f482aad570f7d5965595a9c2a8#diff-38d875e1d50b3158dc96404b65275a54R692)) but it's also possible for the extension author to put `_` in the extension ID (aka "category") or the opcode.

IMO it would make sense to forbid that and enforce it during load by replacing `_` with something else or even removing it, since that's only meant as the machine-readable ID anyway. That said, the conversion between "extended opcode" and category+opcode pair is effectively part of the project JSON spec, so it's not safe to change it unnecessarily.

Another approach would be to build a flat lookup table indexed by extended_opcode, which would mean that we wouldn't need to `split` here and extra `_` characters wouldn't cause any trouble.

Do you think adding the category to MonitorRecord would be a better approach? I'm not very familiar with the monitor framework...

Reply:
I think it's okay to split on `_` here like this, since we rely on that same method for determining the extension in both the sb2 and sb3 deserializers. I *do* think that `.split` here should be replaced with the `splitFirst` function defined in `src/util/StringUtil.js` so that the opcode portion is resilient against extra `_`s.

Reply:
MonitorRecord is just a javascript object with a description of what keys are expected to be in the object. You could add 'category' if you want to have it conveniently available separate from the opcode, or get it from the opcode, whatever's easier.